content/zh/docs/devops-user-guide/how-to-integrate/sonarqube.md

```diff
@@ -1,220 +1,243 @@
+   export NODE_IP=$(kubectl get nodes --namespace kubesphere-devops-system -o jsonpath="{.items[0].status.addresses[0].address}")
+   echo http://$NODE_IP:$NODE_PORT
+   ```
 
-2. 您可以得到以下输出，获取您 Jenkins 的端口号。
+2. 您可以获得以下输出，获取 Jenkins 的端口号。
 
-```bash
-http://10.77.1.201:30180
-```
+   ```bash
+   http://10.77.1.201:30180
+   ```
 
-3. 使用地址 `http://10.77.1.201:30180` 访问 Jenkins。安装 KubeSphere 时，默认情况下也会安装 Jenkins 仪表板。 Jenkins 配置了KubeSphere LDAP，这意味着您可以直接使用 KubeSphere 帐户（例如 admin/P@88w0rd）登录 Jenkins。 有关配置 Jenkins 的更多信息，请参阅 [Jenkins 系统设置](../../../devops-user-guide/how-to-use/jenkins-setting/)。
+3. 请使用地址 `http://Public IP:30180` 访问 Jenkins。安装 KubeSphere 时，默认情况下也会安装 Jenkins 仪表板。此外，Jenkins 还配置有 KubeSphere LDAP，这意味着您可以直接使用 KubeSphere 帐户（例如 `admin/P@88w0rd`）登录 Jenkins。有关配置 Jenkins 的更多信息，请参见 [Jenkins 系统设置](../../../devops-user-guide/how-to-use/jenkins-setting/)。
 
-![jenkins-login-page](/images/docs/devops-user-guide-zh/integrate-sonarqube-into-pipeline-zh/jenkins-login-page.png)
+   ![Jenkins 登录页面](/images/docs/zh-cn/devops-user-guide/tool-integration/integrate-sonarqube-into-pipelines/jenkins-login-page.png)
 
-{{< notice note >}}
+   {{< notice note >}}
 
-您可能需要设置必要的端口转发规则并打开端口 `30180` 才能访问安全组中的 Jenkins，具体取决于您的实例部署的位置。
+   取决于您的实例的部署位置，您可能需要设置必要的端口转发规则，并在您的安全组中放行端口 `30180`，以便访问 Jenkins。
 
-{{</ notice >}} 
+   {{</ notice >}} 
 
-4. 单击左侧的 **Manage Jenkins**。
+4. 点击左侧的 **Manage Jenkins**。
 
-![manage-jenkins](/images/docs/devops-user-guide-zh/integrate-sonarqube-into-pipeline-zh/manage-jenkins.png)
+   ![管理 Jenkins](/images/docs/zh-cn/devops-user-guide/tool-integration/integrate-sonarqube-into-pipelines/manage-jenkins.png)
 
-5. 向下翻页找到并单击 **Configure System**。
+5. 向下翻页找到并点击 **Configure System**。
 
-![configure-system](/images/docs/devops-user-guide-zh/integrate-sonarqube-into-pipeline-zh/configure-system.png)
+   ![configure-system](/images/docs/zh-cn/devops-user-guide/tool-integration/integrate-sonarqube-into-pipelines/configure-system.png)
 
-6. 导航到 **SonarQube servers**，然后单击 **Add SonarQube**。
+6. 搜寻到 **SonarQube servers**，然后点击 **Add SonarQube**。
 
-![add-sonarqube](/images/docs/devops-user-guide-zh/integrate-sonarqube-into-pipeline-zh/add-sonarqube.png)
+   ![添加 SonarQube](/images/docs/zh-cn/devops-user-guide/tool-integration/integrate-sonarqube-into-pipelines/add-sonarqube.png)
 
-7. 输入 **Name**，**Server URL** (`http://10.77.1.201:31377`)和 **Server authentication token**（SonarQube 管理管理员 token）。 单击**Apply**完成。
+7. 输入 **Name**、**Server URL** (`http://Node IP:port`) 和 **Server authentication token**（SonarQube 管理管理员令牌）。点击 **Apply** 完成操作。
 
-![sonarqube-jenkins-settings](/images/docs/devops-user-guide-zh/integrate-sonarqube-into-pipeline-zh/sonarqube-jenkins-settings.png)
+   ![sonarqube-jenkins-settings](/images/docs/zh-cn/devops-user-guide/tool-integration/integrate-sonarqube-into-pipelines/sonarqube-jenkins-settings.png)
 
-### 步骤 6: 将 sonarqubeUrl 添加到 KubeSphere 控制台
+### 步骤 6：将 sonarqubeUrl 添加到 KubeSphere 控制台
```

Comment:
sonarqubeUrl
->
sonarqubeURL

Change the English source text to "sonarqubeURL"